src/components/workitems/workItemHeader.tsx

```diff
@@ -1,59 +1,62 @@
-import { ILocationService } from "azure-devops-extension-api";
+
+import * as React from "react";
 import { IWorkItemFormNavigationService } from "azure-devops-extension-api/WorkItemTracking";
 import * as DevOps from "azure-devops-extension-sdk";
-import * as React from "react";
 import { IWorkItem } from "../../model/workitem";
 import { WorkItemTypeIcon } from "./typeIcon";
 import "./workItemHeader.scss";
 
+
+
 export interface IWorkItemHeaderProps {
     workItem: IWorkItem;
+
 }
+export const WorkItemHeader: React.FC<IWorkItemHeaderProps> = (props) => {
+
+    const { workItem: { id, project, title, workItemType, icon, color, } } = props;
+    const [currentHostName, setCurrentHostName] = React.useState<string>("#")
+
+     const getCurrentHost = async () => {
+        const locationService = await DevOps.getHost()
+        setCurrentHostName(locationService.name)
+    }
 
-type MyState = { url: string };
-export class WorkItemHeader extends React.Component<IWorkItemHeaderProps, MyState > {
-    private workItemUrl = "#";
-    constructor(props:any) {
-        super(props);
-        this.state = {url: ""};
-      }
-
-    async componentDidUpdate() {
-        const locationService = await DevOps.getService<ILocationService>("ms.vss-features.location-service");
-        this.workItemUrl = await locationService.routeUrl(
-            "ms.vss-work-web.work-items-form-route-with-id",
-            { project: this.props.workItem.project, id: this.props.workItem.id.toString() });
-            this.setState({url: this.workItemUrl})
-           }
-    
-       render(): JSX.Element {
-        const {
-            workItem: { id, project, title, workItemType, icon, color }
-        } = this.props;
-
-         const openWi = async (ev:any)=>{
-            if(!ev.ctrlKey && !ev.metaKey &&!ev.altKey && !ev.shiftKey){
+    React.useEffect(() => {
```

Comment:
The current host name logic could be extracted into a custom hook, making this a little cleaner:

```ts
import React from "react";
import * as DevOps from "azure-devops-extension-sdk";

function useCurrentHostName() {
    const [currentHostName, setCurrentHostName] = React.useState<string>("#");

    React.useEffect(() => {
        const locationService = DevOps.getHost();
        setCurrentHostName(locationService.name);
    }, []);

    return currentHostName;
}

export default useCurrentHostName;
````

```ts
const currentHostName = useCurrentHostName();
````

Also, at least according to the typings, `DevOps.getHost()` doesn't need to be awaited.